artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/PagingStoreImpl.java

```diff
@@ -1096,6 +1096,7 @@ public boolean checkMemory(boolean runOnFailure, Runnable runWhenAvailableParame
          if (isFull()) {
             if (runOnFailure && runWhenAvailable != null) {
                addToBlockList(runWhenAvailable, blockedCallback);
+               pagingManager.addBlockedStore(this);
```

Comment:
The use of this same call lower down the class is gated by `if (usingGlobalMaxSize || pagingManager.isDiskFull()) {``, does this need to be?

It is also surrounded by various other double-checks around racing being not-full again, it seems hard to reconcile that path needing it but this one not.

I cant quite wrap my head around this so think @clebertsuconic might need to chip in here.

Reply:
me neither. Need @clebertsuconic's expertise. :)

Reply:
PR updated. The code path is really processing many conditions, not sure what's impact of that.